unix/main.c

```diff
@@ -416,7 +416,7 @@ int main(int argc, char **argv) {
             } else if (strcmp(argv[a], "-v") == 0) {
                 mp_verbose_flag++;
             } else if (strncmp(argv[a], "-O", 2) == 0) {
-                if (isdigit(argv[a][2])) {
+                if (isdigit((byte)argv[a][2])) {
```

Comment:
Well, from that "fix", the issue is quite obvious - the common way to implement isdigit() is by subscripting a table of flags, which of course kabooms if char is "negative". But then it's quite a buggy isdigit(), which requires such "fix" applied to each its invocation, and it's isdigit() which requires fixing.

But deeper problem with this PR is that it packs unrelated changes, and number of commits in it is ~30, so it's hard to reason what's done for what.